src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1983,38 +1983,49 @@ public void RemoveProjectOutputPath_NoLock(SolutionChangeAccumulator solutionCha
 
         private void RefreshMetadataReferencesForFile(object sender, string fullFilePath)
```

Comment:
Frankly, this is why I like "async void" -- it makes the evil clear! I can rename this method if that'd help though.

The waiter here is also of limited value, outside of unit tests -- the file change service still can't be "waited" for in real VS in any meaningful way.

Reply:
> Frankly, this is why I like "async void" -- it makes the evil clear! I can rename this method if that'd help though.

Our normal pattern for cases like this would be `Enqueue*` or `Schedule*` similar.

Reply:
Although neither of those seem great _either_ since this isn't doing anything with a queue (or at least not obviously.)

Honestly, is async void really the right option here? It's an event handler, so that's why we're forced to the signature, but we're still wanting async...

Reply:
Async void isn't part of the signature, so it doesn't resolve the original concern at all (callers cannot tell that the effects are asynchronous).

Reply:
> Although neither of those seem great either since this isn't doing anything with a queue (or at least not obviously.)

Scheduling work in the thread pool would count as queueing

Reply:
@sharwell So in this case there _are_ no callers; it's just the event handler. So there's nobody to inform.

> Scheduling work in the thread pool would count as queueing

But this isn't doing any scheduling either...it's just doing it.

Reply:
Gonig with "StartRefreshing...".